File: tools/cmux-home/typescript/tests/summary.test.ts
import { describe, expect, test } from "bun:test";
import { renderSummary } from "../src/summary";
import { parseHomeState } from "../src/state";

describe("summary output", () => {
  test("prints deterministic once output", () => {
    const state = parseHomeState({
      sessions: [
        {
          agent: "codex",
          sessionId: "codex-1",
          status: "working",
          title: "Build home",
          cwd: "/repo",
        },
        {
          agent: "claude",
          sessionId: "claude-1",
          status: "awaiting",
          title: "Review plan",
          branch: "feat-home",
        },
      ],
    });

    expect(renderSummary(state)).toBe(`cmux home
sessions: total=2
adapters: claude=1 codex=1 opencode=0 pi=0
statuses: awaiting=1 working=1 completed=0

awaiting:
- claude/claude-1 "Review plan" branch=feat-home resume="claude --resume claude-1" gaps=2

working:
- codex/codex-1 "Build home" cwd=/repo resume="cd /repo && codex resume codex-1" gaps=2

task prompt: describe the next task for an agent
`);
  });
});

File: tools/cmux-home/go/internal/home/summary_test.go
package home

import "testing"

func TestSummaryIsDeterministic(t *testing.T) {
	state := HomeState{
		Sessions: []Session{
			{ID: "done-1", Adapter: "pi", Status: "completed", Title: "Finished", CWD: "/tmp/pi"},
			{ID: "run-2", SessionID: "codex-native", Adapter: "codex", Status: "working", Title: "Beta", CWD: "/tmp/codex"},
			{ID: "run-1", Adapter: "claude", Status: "working", Title: "Alpha", CWD: "/tmp/claude"},
		},
		Tasks: []Task{{ID: "task", Title: "Queued"}},
	}

	want := `cmux home
adapters: claude=1 codex=1 opencode=0 pi=1
working: 2
  claude run-1 Alpha [/tmp/claude]
  codex codex-native Beta [/tmp/codex]
completed: 1
  pi done-1 Finished [/tmp/pi]
selected: run-1
resume: cd '/tmp/claude' && 'claude' '--resume' 'run-1'
task prompt: 1 queued
`
	if got := Summary(state); got != want {
		t.Fatalf("Summary mismatch\n--- got ---\n%s--- want ---\n%s", got, want)
	}
}
